subsys/bluetooth/host/conn.c

```diff
@@ -377,15 +377,23 @@ static void bt_acl_recv(struct bt_conn *conn, struct net_buf *buf,
 static void wait_for_tx_work(struct bt_conn *conn)
 {
 #if defined(CONFIG_BT_CONN_TX)
-	struct k_work_sync sync;
+	LOG_DBG("conn %p", conn);
 
-	/* API docs mention undefined behavior if syncing on work item from wq
-	 * execution context.
-	 */
-	__ASSERT_NO_MSG(k_current_get() != &k_sys_work_q.thread);
+	if (IS_ENABLED(CONFIG_BT_RECV_WORKQ_SYS)) {
+		__ASSERT_NO_MSG(k_current_get() == &k_sys_work_q.thread);
 
-	k_work_submit(&conn->tx_complete_work);
-	k_work_flush(&conn->tx_complete_work, &sync);
+		tx_notify(conn);
+	} else {
+		struct k_work_sync sync;
+
+		/* API docs mention undefined behavior if syncing on work item
+		 * from wq execution context.
+		 */
+		__ASSERT_NO_MSG(k_current_get() != &k_sys_work_q.thread);
```

Comment:
Consider using the getter `k_work_queue_thread_get`

Reply:
no one uses it but ok